osgp/platform/osgp-adapter-kafka-distributionautomation/src/test/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/GridMeasurementPublishedEventConverterTest.java

```diff
@@ -0,0 +1,98 @@
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.opensmartgridplatform.adapter.kafka.da.avro.AccumulationKind;
+import org.opensmartgridplatform.adapter.kafka.da.avro.Analog;
+import org.opensmartgridplatform.adapter.kafka.da.avro.AnalogValue;
+import org.opensmartgridplatform.adapter.kafka.da.avro.GridMeasurementPublishedEvent;
+import org.opensmartgridplatform.adapter.kafka.da.avro.MeasuringPeriodKind;
+import org.opensmartgridplatform.adapter.kafka.da.avro.Name;
+import org.opensmartgridplatform.adapter.kafka.da.avro.PhaseCode;
+import org.opensmartgridplatform.adapter.kafka.da.avro.UnitMultiplier;
+import org.opensmartgridplatform.adapter.kafka.da.avro.UnitSymbol;
+
+class GridMeasurementPublishedEventConverterTest {
+
+    private final DistributionAutomationMapper mapper = new DistributionAutomationMapper();
+
+    @Test
+    void testConvertMeasurement() {
+        final String measurement = "TST-01; 220.1; 220.2; 220.3; 5.1; 5.2; 5.3; 7.1; 7.2; 7.3;";
+        final GridMeasurementPublishedEvent event = this.mapper.map(measurement, GridMeasurementPublishedEvent.class);
+        final List<Analog> measurements = event.getMeasurements();
+
+        assertThat(measurements).usingElementComparatorIgnoringFields("mRID").isEqualTo(this.expectedMeasurements());
+    }
+
+    @Test
+    void testConvertCongestion() {
+        final String measurement = "TST-01; 5.1; 5.2; 5.3; 7.1; 7.2; 7.3;";
+        final GridMeasurementPublishedEvent event = this.mapper.map(measurement, GridMeasurementPublishedEvent.class);
+        final List<Analog> measurements = event.getMeasurements();
+
+        assertThat(measurements).usingElementComparatorIgnoringFields("mRID")
+                .isEqualTo(this.expectedCurrentMeasurements());
+    }
+
+    @Test
+    void testSomeOtherString() {
+        final String someOtherString = "TST-01";
+        final GridMeasurementPublishedEvent event = this.mapper.map(someOtherString,
+                GridMeasurementPublishedEvent.class);
+
+        assertThat(event).isNull();
+
+    }
+
+    private List<Analog> expectedMeasurements() {
+        final List<Analog> measurements = new ArrayList<>();
+        measurements.add(new Analog("TST-01", null, AccumulationKind.none, MeasuringPeriodKind.none, PhaseCode.none,
+                UnitMultiplier.none, UnitSymbol.V, new ArrayList<Name>(),
+                Arrays.asList(new AnalogValue(220.1f, null, null))));
+        measurements.add(new Analog("TST-01", null, AccumulationKind.none, MeasuringPeriodKind.none, PhaseCode.none,
+                UnitMultiplier.none, UnitSymbol.V, new ArrayList<Name>(),
+                Arrays.asList(new AnalogValue(220.2f, null, null))));
+        measurements.add(new Analog("TST-01", null, AccumulationKind.none, MeasuringPeriodKind.none, PhaseCode.none,
+                UnitMultiplier.none, UnitSymbol.V, new ArrayList<Name>(),
+                Arrays.asList(new AnalogValue(220.3f, null, null))));
```

Comment:
Consider extracting a method for expectedVoltageMeasurements as well in order to improve readability.

Reply:
done